src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -914,25 +924,253 @@ void Pane::_Split(SplitState splitType, const GUID& profile, const TermControl&
+// - widthOrHeight: if true operates on width, otherwise on height.
+// - sizeNode: a layouting node that corresponds to this pane.
+// Return Value:
+// - <none>
+void Pane::_AdvanceSnappedDimension(const bool widthOrHeight, LayoutSizeNode& sizeNode) const
+{
+    if (_IsLeaf())
+    {
+        if (sizeNode.isMinimumSize)
+        {
+            // If the node is of its minimum size, this size might not be snapped,
+            // so snap it upward. It might however be snapped, so add 1 to make
+            // sure it really increases (not really required but to avoid surprises).
+            sizeNode.size = _SnapDimension(widthOrHeight, sizeNode.size + 1).second;
+        }
+        else
+        {
+            const auto cellSize = _control.CharacterDimensions();
+            sizeNode.size += widthOrHeight ? cellSize.Width : cellSize.Height;
+        }
+    }
+    else
+    {
+        // The given node often has next possible (advanced) values already
+        // cached by the previous advance operation. If we're the first one,
+        // we need to calculate them now.
+        if (sizeNode.nextFirstChild == nullptr)
+        {
+            sizeNode.nextFirstChild.reset(new LayoutSizeNode(*sizeNode.firstChild));
+            _firstChild->_AdvanceSnappedDimension(widthOrHeight, *sizeNode.nextFirstChild);
+        }
+        if (sizeNode.nextSecondChild == nullptr)
+        {
+            sizeNode.nextSecondChild.reset(new LayoutSizeNode(*sizeNode.secondChild));
+            _secondChild->_AdvanceSnappedDimension(widthOrHeight, *sizeNode.nextSecondChild);
+        }
+
+        const auto nextFirstSize = sizeNode.nextFirstChild->size;
+        const auto nextSecondSize = sizeNode.nextSecondChild->size;
+
+        bool advanceFirst; // Whether to advance first or second child
+        if (_splitState == (widthOrHeight ? SplitState::Horizontal : SplitState::Vertical))
+        {
+            // If we're growing along separator axis, choose the child that
+            // wants to be smaller than the other.
+            advanceFirst = nextFirstSize < nextSecondSize;
+        }
+        else
+        {
+            // If we're growing perpendicularly to separator axis, choose
+            // the child so that their size ratio is closer to the currently
+            // maintained (so that the relative separator position is closer
+            // to the _desiredSplitPosition field).
+
+            const auto firstSize = sizeNode.firstChild->size;
+            const auto secondSize = sizeNode.secondChild->size;
+
+            // Because we relay on equality check these calculations have to be
+            // immune to floating point errors.
+            const auto deviation1 = nextFirstSize - (nextFirstSize + secondSize) * _desiredSplitPosition;
+            const auto deviation2 = -1 * (firstSize - (firstSize + nextSecondSize) * _desiredSplitPosition);
+            advanceFirst = deviation1 <= deviation2;
+        }
+
+        if (advanceFirst)

```

Comment:
Each `_AdvanceSnappedDimension` call advances excellently one of the pane's children (unless it's a leaf) and here it decides which one will it be. And according to my convention `advanceFirst` should really be `advanceFirstOrSecond`.

If your talking about the general idea, then admittedly it's not the very simplest thing in the world. Do you think this would benefit from some further explanation?